bilby/core/sampler/dynesty.py

```diff
@@ -706,15 +708,27 @@ def _run_external_sampler_with_checkpointing(self):
                 ).total_seconds()
             if last_checkpoint_s > self.check_point_delta_t:
                 self.write_current_state()
+                self._add_live()
                 self.plot_current_state()
-        if getattr(self.sampler, "added_live", False):
-            self.sampler._remove_live_points()
+                self._remove_live()
 
+        self._remove_live()
+        if "add_live" in sampler_kwargs:
+            sampler_kwargs["add_live"] = True
```

Comment:
Could you help me understand why we need these lines?

Following the call to `finalize_sampler_kwargs`, `add_live` should always be false. Is this reverting that? If so, what if `add_live` was `False` to begin with?

Reply:
Yes, that's what I meant. I think it from the original kwargs makes sense.